unified-prototype/unified-plugin/plugin-android/src/main/java/org/gradle/api/experimental/android/library/StandaloneAndroidLibraryPlugin.java

```diff
@@ -52,6 +65,14 @@ public static void linkDslModelToPlugin(Project project, AndroidLibrary dslModel
             ifPresent(dslModel.getMinSdk(), defaultConfig::setMinSdk);
             return null;
         });
+        android.compileOptions(compileOptions -> {
+            // Up to Java 11 APIs are available through desugaring
+            // https://developer.android.com/studio/write/java11-minimal-support-table
+            compileOptions.setSourceCompatibility(JavaVersion.VERSION_11);
+            compileOptions.setTargetCompatibility(JavaVersion.VERSION_11);
+            compileOptions.setCoreLibraryDesugaringEnabled(!dslModel.getDependencies().getCoreLibraryDesugaring().getDependencies().get().isEmpty());
```

Comment:
I don't think we want to do it this way because I'm not sure about the ordering of this. It seems like this might get evaluated before the project's build file.

Can we either always set this to true/false or make sure this is evaluated late?